app/javascript/src/local_level.tsx

```diff
@@ -10,7 +10,31 @@ const TILE_CLASSES = {
   2: "floor",
 };
 
-class Tile extends React.Component {
+interface IPlayer {
+  x: number;
+  y: number;
+};
+
+interface ITile {
+  x:          number;
+  y:          number;
+  tile:       string;
+  visit_path: string;
+  monsters: {
+    type:   string;
+    health: number;
+  }[];
+};
+
+interface TileProps {
+  parent: LocalLevel;
+  player: IPlayer;
+  tile:   ITile;
+}
+
+class Tile extends React.PureComponent<TileProps> {
+  props: TileProps;
+
   handleVisit = (e) => {
     let path = this.props.tile.visit_path;
```

Comment:
Just a side note and not relevant to this PR really. I'm a huge fan of destructuring `const {tile, parent} = this.props` which helps to avoid having `this.props...` all over the place.